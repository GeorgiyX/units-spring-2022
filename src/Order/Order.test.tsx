import React from 'react';
import 'jest';
import {shallow, configure} from 'enzyme';
import {OrderComponent} from './Order';
import {fakeOrders} from '../data/fakeOrders';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const mocGetData: jest.Mock = jest.fn()
	.mockReturnValue('aa')
	.mockReturnValue('bb')
	.mockReturnValue('cc');

describe('Order.tsx', () => {
	let wrapper;

	it('no items (Сбереги Мега Маркер)', () => {
		wrapper = shallow(<OrderComponent order={fakeOrders[0]}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('items (Alihandro Express)', () => {
		wrapper = shallow(<OrderComponent order={fakeOrders[1]}/>);
		expect(wrapper).toMatchSnapshot();
	});

	it('items (Alihandro Express)', () => {
		wrapper = shallow(<OrderComponent order={{date: 0}}/>);
		expect(wrapper).toMatchSnapshot();
	});
});
