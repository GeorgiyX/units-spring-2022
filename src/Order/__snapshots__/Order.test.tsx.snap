// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.tsx items (Alihandro Express) 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      9 декабря, вс, 2018 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;

exports[`Order.tsx items (Alihandro Express) 2`] = `""`;

exports[`Order.tsx no items (Сбереги Мега Маркер) 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Сбереги Мега Маркер
    </span>
    <span
      id="date"
    >
      1 мая, пт, 2020 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;

exports[`Order.tsx null data 1`] = `""`;

exports[`Order.tsx simple (Сбереги Мега Маркер) 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Сбереги Мега Маркер
    </span>
    <span
      id="date"
    >
      1 мая, пт, 2020 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div>
      Заказ пуст...
    </div>
  </div>
</div>
`;
